common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'rpm_qa', 'rpm_Va', 'rpm_etc_Va', 'chkconfig'".
+                    return 1
+                    ;;
+            esac
+            ;;
+        "filelist")
+            case $key in
+                "allmyfiles")
+                    _get_cached_command $PREUPG_ALLMYFILES
+                    ;;
+                "executable")
+                    _get_cached_command $PREUPG_EXECUTABLES
+                    ;;
+                "rpmtrackedfiles")
+                    _get_cached_command $PREUPG_RPMTRACKEDFILES
+                    ;;
+                "rpmrhsignedfiles")
+                    _get_cached_command $PREUPG_RPM_RHSIGNED
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'executable', 'allmyfiles', 'rpmtrackedfiles', 'rpmrhsignedfiles'."
+                    return 1
+            esac
+            ;;
+        "info")
+            ;;
+        *)
+            log_error "Unknown command $command. Supported are 'file', 'command', filelist', 'info'."
+            return 1
+            ;;
+    esac
+
+}
+
+add_postupgrade()
+{
+    # Add postupgrade script to /root/preupgrade/postupgrade.d directory
+    # File as first parameter has to exists in modules directory
+    # Usage:    add_postupgrade file
+
+    script_name=$1
+    postupgrade_name="$PREUPG_POSTUPGRADE_DIR/`basename $script_name`"
+    [ ! -f "$script_name" ] && log_error "$script_name does not exist." && exit_error
+    [ -f "$postupgrade_name" ] && log_warning "$script_name already exists in $PREUPG_POSTUPGRADE_DIR"
+    cp -af $script_name $postupgrade_name
+}
+
+add_manual_postupgrade()
+{
+    # Add postupgrade script to /root/preupgrade/noauto_postupgrade.d directory
+    # File as first parameter has to exists in modules directory
+    # Usage:    add_manual_postupgrade file
+
+    script_name=$1
+    postupgrade_name="$NOAUTO_POSTUPGRADE_D/`basename $script_name`"
+    [ ! -f "$script_name" ] && log_error "$script_name does not exist." && exit_error
+    [ -f "$postupgrade_name" ] && log_warning "$script_name already exists in $NOAUTO_POSTUPGRADE_D"
+    cp -af $script_name $postupgrade_name
+}
+
+add_to_kickstart_readme()
+{
+    # Add filename and description to /root/preupgrade/kickstart/README file
+    # Filename as first parameter which will be inserted into README file
+    # Description of filename as second parameter which will be inserted into README file.
+    # Format in README file is:
+    # * <filename> - <description>
+    # Usage:    add_to_kickstart_readme <filename> <description>
+    filename=$1
+    description=$2
+    [ -z "$1" ] && log_error "Missing filename parameter."
+    [ -z "$2" ] && log_error "Missing description of filename to kickstart README."
+    echo " * $filename - $description" >> $KICKSTART_README
+}
```

Comment:
IIUC, user will need two steps: adding file to KS (`add_kickstart_dir())`, and describing the file `(add_kickstart_readme())`.

Shouldn't one step be enough?

With a bit of shuffling, one variadic function can handle even more rich scenarios.  E.g. (naive code):

    keep_for_kickstart() {
        #
        # Store and keep file(s) for kickstart installation
        #
        # Usage:
        #
        #     keep_for_kickstart <description> <file>...
        #
        # Keep one or more <file>s for kickstart installation. <description> is mandatory and
        # will appear in README provided along with stored files.
        #
        local description=$1; shift
        local item
        local list_open=false
        {
            echo
            echo -n " * "
            for item in "$@"; do 
                # some checks...
                cp -ar $item $KS_STORAGE
                $list_open && echo -n ", "            
                echo -n "$(basename "$item")"; list_open=true
            done
            echo " - $description"
        } >> $KS_FILE
    }

Note that with above API, user can not just do both storing and documenting in one step, but also share description among multiple files.